src/libcore/iter/iterator.rs

```diff
@@ -88,7 +88,7 @@ fn _assert_is_object_safe(_: &dyn Iterator<Item=()>) {}
     message="`{Self}` is not an iterator"
 )]
 #[doc(spotlight)]
-#[must_use]
+#[must_use = "iterator adaptors are lazy and do nothing unless consumed"]
```

Comment:
The existing docs use "iterator adapter", so I think we want to use that spelling instead.

Reply:
It seems that "iterator adaptors" is used as #[must_use] message: https://github.com/rust-lang/rust/blob/master/src/libcore/iter/mod.rs#L407

Reply:
Hm, that one should probably be changed too, then.

Reply:
In the Rust code on GitHub, [iterator adaptors](https://github.com/search?l=&p=1&q=iterator+adaptors+language%3ARust&type=Code) seems to be used more than [iterator adapter](https://github.com/search?l=&p=1&q=iterator+adapter+language%3ARust&type=Code).

IMO, this should be handled by another PR that like "Unify the name of the iterator adapter".

Reply:
My preference is to just drop "adaptors" in favor of "iterators are lazy and do...", since it's more direct.